http-gateways/IPLD_GATEWAY.md

```diff
@@ -0,0 +1,196 @@
+# IPLD Gateway Specification
+
+**Authors**:
+
+- Mauve Signweaver ([@RangerMauve](https://github.com/RangerMauve))
+
+----
+
+**Abstract**
+
+IPLD Gateway is an extension of [PATH_GATEWAY.md](./PATH_GATEWAY.md) that enables lower level interaction with IPLD data structures under a specific path.
+
+This document describes the delta between [PATH_GATEWAY.md](./PATH_GATEWAY.md) and this gateway type.
+
+Summary:
+
+- Adds a new `/ipld/{cid}[/{segments}][?{params}]` subpath to the gateway
+- Defines a specification for parsing out extra parameters for individual path segments.
+- Describes how to map to and from `ipld://` URLs
+
+## HTTP API
+
+### `GET /ipld/{cid}[/{segments}][?{params}]`
+
+Resolve IPLD paths to some data.
+
+The path segments will be traversed with any parameters used to transform data along the way.
+
+The `format` query string parameter, or the `Accept` request header can be used to control the format which will be used to return the data.
+
+By default, data will be returned as `dag-json`
```

Comment:
first time we mention dag-json, good to link to specs and make it clear implementations need to support both  text and binary.

```suggestion
By default, data will be returned as [DAG-JSON](https://ipld.io/specs/codecs/dag-json/).  Implementations MUST also support [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/) as an opt-in (requested by the client).
```